org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentLifeCycleHandler.java

```diff
@@ -302,6 +304,7 @@ public void handleClosed(DidCloseTextDocumentParams params) {
 	public void handleSaved(DidSaveTextDocumentParams params) {
 		String uri = params.getTextDocument().getUri();
 		ICompilationUnit unit = JDTUtils.resolveCompilationUnit(uri);
+		unit = checkPackageDeclaration(uri, unit);
```

Comment:
Can you add some documentation why we are doing this, possibly a link to the GH issue

Reply:
I have added it.